tests/test_specifiers.py

```diff
@@ -976,3 +976,17 @@ def test_comparison_canonicalizes(self, left, right):
     def test_comparison_non_specifier(self):
         assert SpecifierSet("==1.0") != 12
         assert not SpecifierSet("==1.0") == 12
+
+    @pytest.mark.parametrize(
+        ("version", "specifier", "expected"),
+        [
+            ("1.0.0+local", "==1.0.0", True),
+            ("1.0.0+local", "!=1.0.0", False),
+            ("1.0.0+local", "<=1.0.0", True),
+            ("1.0.0+local", ">=1.0.0", True),
```

Comment:
These are the two cases important to this PR. The other four are already passing and just added for completeness.

The first one (`Version("1.0.0+local") in SpecifierSet("<=1.0.0")`) is the failing test case.

The second is already passing due to a fluke: because a version with a local version labels is always considered greater than the same version without it, the "greater than" portion of "greater than or equal to" causes it to be true. This PR makes it pass due to the "equal to" portion causing it to be true instead, which is more correct, but indistinguishable in this test.